dScripts/EquipmentScripts/PersonalFortress.cpp

```diff
@@ -2,49 +2,56 @@
 #include "GameMessages.h"
 #include "SkillComponent.h"
 #include "DestroyableComponent.h"
+#include "ControllablePhysicsComponent.h"
 #include "EntityManager.h"
 
 void PersonalFortress::OnStartup(Entity* self) {
 	auto* owner = self->GetOwner();
 	self->AddTimer("FireSkill", 1.5);
-	GameMessages::SendSetStunned(owner->GetObjectID(), PUSH, owner->GetSystemAddress(), LWOOBJID_EMPTY,
-		true, true, true, true, true, true, true, true, true
-	);
 
-	auto* destroyableComponent = owner->GetComponent<DestroyableComponent>();
+	auto* destroyableComp = owner->GetComponent<DestroyableComponent>();
+	if (destroyableComp) destroyableComp->SetStatusImmunity(
+			eStateChangeType::PUSH,
+			true, true, true, true, true, false, true, false, false
+		);
 
-	if (destroyableComponent != nullptr) {
-		destroyableComponent->PushImmunity();
-	}
+	auto* controllablePhysicsComp = owner->GetComponent<ControllablePhysicsComponent>();
+	if (controllablePhysicsComp) controllablePhysicsComp->SetStunImmunity(
+			eStateChangeType::PUSH, LWOOBJID_EMPTY,
+			true, true, true, true, true, true
+		);
+
+	GameMessages::SendSetStunned(owner->GetObjectID(), eStateChangeType::PUSH, owner->GetSystemAddress(), LWOOBJID_EMPTY,
+		true, true, true, true, true, true, true, true, true
+	);
```

Comment:
What is the motivation for this change?

Reply:
Okay so personal fortress is the one script I will leave in this PR since I completely broke it.
I removed the two function calls it was using which was basically all the script did beside a cast skill, so I properly implemented it as a test to the immunity code